docs/tutorials/data_provider.md

```diff
@@ -68,6 +68,8 @@ A key feature of ICU4X is the ability to download data dynamically, allowing cli
 
 Dynamic data loading can currently be performed in user code. A future core library API may provide this functionality; please submit feedback in [#2985](https://github.com/unicode-org/icu4x/issues/2985).
 
+The following example loads additional locales bucketed by language. This means that different script and regional variants of the same language are assumed to be in the same dynamically loaded data file. However, clients should choose a dynamic loading strategy that works best for them.
```

Comment:
> However, clients should choose a dynamic loading strategy that works best for them

The only strategy you're demonstrating is this (bad) one, clients won't know what "works best for them".

I'd also prefer to see this as a code comment where you do the bucketing, that'll be harder to miss when copy-pasting.

Reply:
I don't totally agree that it's always a bad strategy. Dynamic loading doesn't need to be as strict as compile-time slicing. It's okay if you dynamically load a little more data than you need. The issue with language packs that target only one script is that there are parts of the fallback infrastructure that cross scripts, and also, when downloading the pack, you need to know which script, which is nontrivial.

In other words, loading all the data for a language is almost always going to _work_, even if it is in excess, and it is _easy to calculate_.

I'll add another comment in the code section.

_Note: I acknowledge that there is a CLDR parent relationship that crosses language boundaries, hi-Latn to en-IN, which is annoying. I'm ignoring this on the basis that apps are highly likely to have English symbols available by the time hi-Latn is requested._